modules/system/assets/js/framework.js

```diff
@@ -41,23 +41,36 @@ if (window.jQuery.request !== undefined) {
         $triggerEl.trigger(_event, context)
         if (_event.isDefaultPrevented()) return
 
-        var data = [$form.serialize()],
-            loading = options.loading !== undefined ? options.loading : null,
+        var loading = options.loading !== undefined ? options.loading : null,
             isRedirect = options.redirect !== undefined && options.redirect.length,
-            useFlash = options.flash !== undefined
+            useFlash = options.flash !== undefined,
+            sendFiles = $el.is(':file') || ($el[0] == $form[0] && $form.attr('enctype') == 'multipart/form-data') || options.sendFiles,
+            data = $triggerEl.find(':input').addBack(':input').serializeArray()
```

Comment:
When using pure JS you can call `$.request('onEvent')` rather than `$('#form').request('onEvent')` (see https://octobercms.com/docs/ajax/javascript-api#javascript-examples for usage without a form). My changes ended up collecting all the form elements in the page and sending them in the request which is not current AJAX API behaviour.